ArduPlane/altitude.cpp

```diff
@@ -200,7 +200,7 @@ void Plane::set_target_altitude_current(void)
 #if AP_TERRAIN_AVAILABLE
     // also record the terrain altitude if possible
     float terrain_altitude;
-    if (g.terrain_follow && terrain.height_above_terrain(terrain_altitude, true)) {
+    if (g.terrain_follow && terrain.height_above_terrain(terrain_altitude, true) && !nonauto_terrain_disable) {
```

Comment:
I would prefer we replace instances of g.terrain_follow with terrain_follow_enabled() function call, and make that function check the flight mode before applying nonauto_terrain_disable
Even if set_target_altitude_current() is currently only used for non-auto modes, using it in this way is fragile

Reply:
done